File: signalinsight/api_client/projects.py
"""Projects API client module."""

from typing import Any, Dict, List, Optional
from signalinsight.api_client.base import BaseAPIClient


class ProjectsClient:
    def __init__(self, client: BaseAPIClient):
        self.client = client

    def list(self, skip: int = 0, limit: int = 100) -> List[Dict[str, Any]]:
        _, data = self.client.request("GET", "/api/v1/projects", params={"skip": skip, "limit": limit})
        return data or []

    def get(self, project_id: int) -> Dict[str, Any]:
        _, data = self.client.request("GET", f"/api/v1/projects/{project_id}")
        return data

    def create(self, name: str, description: Optional[str] = None) -> Dict[str, Any]:
        _, data = self.client.request("POST", "/api/v1/projects", json_data={"name": name, "description": description})
        return data

    def update(self, project_id: int, name: Optional[str] = None, description: Optional[str] = None, status: Optional[str] = None) -> Dict[str, Any]:
        payload = {}
        if name is not None:
            payload["name"] = name
        if description is not None:
            payload["description"] = description
        if status is not None:
            payload["status"] = status
        _, data = self.client.request("PUT", f"/api/v1/projects/{project_id}", json_data=payload)
        return data

    def delete(self, project_id: int) -> None:
        self.client.request("DELETE", f"/api/v1/projects/{project_id}")

File: frontend/src/api/projects.ts
/**
 * Projects API endpoints client.
 */

import { ApiClient, defaultApiClient } from "./client";
import { Project } from "./types";

export class ProjectsApi {
  constructor(private client: ApiClient = defaultApiClient) {}

  async list(skip = 0, limit = 100): Promise<Project[]> {
    return this.client.request<Project[]>(`/api/v1/projects?skip=${skip}&limit=${limit}`);
  }

  async get(id: number): Promise<Project> {
    return this.client.request<Project>(`/api/v1/projects/${id}`);
  }

  async create(name: string, description?: string): Promise<Project> {
    return this.client.request<Project>("/api/v1/projects", {
      method: "POST",
      body: JSON.stringify({ name, description }),
    });
  }

  async update(id: number, data: Partial<Project>): Promise<Project> {
    return this.client.request<Project>(`/api/v1/projects/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
    });
  }

  async delete(id: number): Promise<void> {
    return this.client.request<void>(`/api/v1/projects/${id}`, {
      method: "DELETE",
    });
  }
}

export const projectsApi = new ProjectsApi();
